hs_app_netCDF/receivers.py

```diff
@@ -191,36 +188,22 @@ def netcdf_pre_delete_file_from_resource(sender, **kwargs):
         # user = kwargs['user']
         # resource_modified(resource, user)
 
-        if del_file_ext == '.nc':
-            # delete the netcdf header text file
+        # delete the netcdf header file or .nc file
+        file_ext = ['.nc', '.txt']
+        if del_file_ext in file_ext:
+            file_ext.remove(del_file_ext)
             for f in ResourceFile.objects.filter(object_id=nc_res.id):
                 ext = os.path.splitext(f.resource_file.name)[-1]
-                if ext == '.txt':
+                if ext in file_ext:
                     f.resource_file.delete()
                     f.delete()
                     break
```

Comment:
If I delete all two files in this function, the system will show error info something like there is no file object to be deleted. This pre_delete_file_from_resource() function is executed before system deletes the user selected resource file from the user interface.